### PR TITLE
Add Layer Categories in map legend (catalog layers)

### PR DESCRIPTION
**Pull request by volterra79**
As in QGIS Desktop TOC Layers, it is possible to manage single categories on layer.

Related to: https://github.com/g3w-suite/g3w-admin/pull/406,  https://github.com/g3w-suite/g3w-admin/issues/407

## G3W-CLIENT (catalog layers)

immagine

## QGIS (table of content)

Screenshot from 2022-08-02 10-48-42

---

**List of changes:**

- add `options = {}` parameter
  - within `core/layers/ImageLayer::getLegendUrl(params={}, options={})` function
  - within `core/layers/legend/LegendService::get({layer, params, options={}}={})` function
  - within `core/layers/legend/WMSLegend({layer, params, options={}})` constructor

- update `gui/catalog/vue/components/layerlegend.vue` methods:
  - add `getSingleLayerLegendCategories()`
  - add `showHideLayerCategory()`
  - update `getLegendSrc()`

### CODE REVIEW

**Comment by Raruto** [comment]
@volterra79 due to https://github.com/g3w-suite/g3w-client/pull/83 it is best to move this to at least within [v3.6](https://github.com/g3w-suite/g3w-client/milestone/5) (otherwise it will become very hard to carry all the new changes within that pull)

File: src/app/core/layers/imagelayer.js [hunk 1]
```diff
@@ -181,13 +181,14 @@ proto.getIconUrlFromLegend = function() {
   })
 };
```
Comment (Raruto) [inline src/app/core/layers/imagelayer.js:203]: Hello

**Review by Raruto — COMMENTED** [review]
@volterra79 could you please also attach a picture of how this thing is handled within the application?

**Comment by volterra79** [comment]
@Raruto 
Screenshot from 2022-10-28 11-00-57

**Pull request merged**